services/resource-coordinator/src/drivers/kubernetes/KubernetesDriver.js

```diff
@@ -0,0 +1,203 @@
+const { BaseDriver } = require('@openintegrationhub/resource-coordinator');
+const uuid = require('uuid/v4');
+const _ = require('lodash');
+const KubernetesRunningApp = require('./KubernetesRunningNode');
+const FlowSecret = require('./FlowSecret');
+
+class KubernetesDriver extends BaseDriver {
+    constructor(config, logger, k8s) {
+        super();
+        this._config = config;
+        this._logger = logger;
+        this._coreClient = k8s.getCoreClient();
+        this._batchClient = k8s.getBatchClient();
+    }
+
+    async createApp(flow, node, envVars, secretEnvVars) {
+        this._logger.info({flow: flow.id}, 'Going to deploy job to k8s');
+        await this._ensureFlowSecret(flow, secretEnvVars);
+        const descriptor = this._buildDescriptor(flow, node, envVars);
+        this._logger.trace(descriptor, 'going to deploy a job to k8s');
+        try {
+            await this._batchClient.jobs.post({body: descriptor});
+        } catch (e) {
+            this._logger.error(e, 'Failed to deploy the job');
+        }
+    }
+
+    async _ensureFlowSecret(flow, secretEnvVars) {
+        const flowSecret = await this._getFlowSecret(flow);
+        if (!flowSecret) {
+            return this._createFlowSecret(flow, secretEnvVars);
+        }
+        return flowSecret;
+    }
+
+    async _getFlowSecret(flow) {
+        try {
+            const result = await this._coreClient.secrets(flow.id).get();
+            return FlowSecret.fromDescriptor(result.body);
+        } catch (e) {
+            if (e.statusCode === 404) {
+                return null;
+            }
+            throw e;
+        }
+    }
+
+    async _createFlowSecret(flow, data) {
+        this._logger.debug('About to create a secret');
+
+        const flowSecret = new FlowSecret({
+            metadata: {
+                name: flow.id,
+                namespace: this._config.get('NAMESPACE'),
+                ownerReferences: [
+                    {
+                        apiVersion: 'elastic.io/v1',
+                        kind: 'Flow',
+                        controller: true,
+                        name: flow.id,
+                        uid: flow.metadata.uid
+                    }
+                ]
+            },
+            data
+        });
+
+        await this._coreClient.secrets.post({
+            body: flowSecret.toDescriptor()
+        });
+
+        this._logger.debug('Secret has been created');
+
+        return flowSecret;
+    }
+
+    async destroyApp(appId) {
+        //TODO wait until job really will be deleted
+        this._logger.info({name: appId}, 'going to undeploy job from k8s');
+        try {
+            await this._batchClient.jobs.delete({
```

Comment:
we need to handle 404 error here. Delete for non-existing entity is not an error.
But any other error is error. So we need to retry deletion in some way. (I assume that resource coordinator works with loop, so reties are already implemented)